packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -0,0 +1,410 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Safe
+import { GnosisSafe as Safe } from "safe-contracts/GnosisSafe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { Guard as IGuard } from "safe-contracts/base/GuardManager.sol";
+import { ExecTransactionParams } from "src/safe/Types.sol";
+
+// Interfaces
+import { ISemver } from "interfaces/universal/ISemver.sol";
+
+/// @title TimelockGuard
+/// @notice This guard provides timelock functionality for Safe transactions
+/// @dev This is a singleton contract. To use it:
+///      1. The Safe must first enable this guard using GuardManager.setGuard()
+///      2. The Safe must then configure the guard by calling configureTimelockGuard()
+contract TimelockGuard is IGuard, ISemver {
+    /// @notice Configuration for a Safe's timelock guard
+    struct GuardConfig {
+        uint256 timelockDelay;
+        bool configured;
+    }
+
+    /// @notice Scheduled transaction
+    struct ScheduledTransaction {
+        uint256 executionTime;
+        bool cancelled;
+        bool executed;
+    }
+
+    /// @notice Mapping from Safe address to its guard configuration
+    mapping(Safe => GuardConfig) public safeConfigs;
```

Comment:
`timelockSafeConfiguration`, as per the specs. So that it can be merged with LivenessModule.